windows/deploy_windows.ps1

```diff
@@ -72,13 +74,14 @@ Function Get-RedirectedUrl {
             $response.Close()
             return
         } catch {
-            if ($attempt -lt 3) {
+            if ($attempt -lt $numAttempts) {
                 Write-Warning "Caught error: $_"
-                Write-Warning "Get-RedirectedUrl: Fetch attempt #${attempt} for $url failed, trying again in 10s"
-                Start-Sleep -Seconds 10
+                Write-Warning "Get-RedirectedUrl: Fetch attempt #${attempt}/${numAttempts} for $url failed, trying again in ${sleepTime}s"
+                Start-Sleep -Seconds $sleepTime
+                $sleepTime = [Math]::Min($sleepTime * 2, 80);
```

Comment:
Go on, put the max into a variable, too.  You know it makes sense...

Reply:
Done.

At some point, all resilient systems seem to consist of huge a collection of workarounds and retries...